sourced/ml/transformers/token_mapper.py

```diff
@@ -0,0 +1,41 @@
+from operator import add
+
+from sourced.ml.transformers.transformer import Transformer
+from pyspark.sql import functions
+
+
+class TokenMapper(Transformer):
+    def __init__(self,  token_parser, prune_size=1, **kwargs):
+        super().__init__(**kwargs)
+        self.token_parser = token_parser
+        self.prune_size = prune_size
+
+    def get_tokens(self, uasts):
+        """
+        Get all tokens from uasts.
+
+        :param uasts: UASTsDataFrame from sourced.engine
+        :return: DataFrame with new tokens column
+        """
+        # TODO (zurk): reimplement uast_ids_to_bag.py using new engine and use it instead
```

Comment:
But you are already using the engine, no?

Reply:
Yes, but we can replace everywhere where we use `UastIds2Bag` class with that one.

Reply:
We cannot because Apollo uses independent extractors which are not the part of the pipeline